lib/linter.js

```diff
@@ -0,0 +1,234 @@
+        let fullPathItem = path.resolve(process.cwd(), path.dirname(configPath), item);
+        if (fullPathModuleId === fullPathItem) {
+          return true;
+        }
+      } else if (item.moduleId) {
+        let fullPathItem = path.resolve(process.cwd(), path.dirname(configPath), item.moduleId);
+        if (fullPathModuleId === fullPathItem) {
+          return item;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  verify(options) {
+    let messages = [];
+    let pendingStatus = this.statusForModule('pending', options.moduleId);
+    let shouldIgnore = this.statusForModule('ignore', options.moduleId);
+
+    if (shouldIgnore) {
+      return messages;
+    }
+
+    let source = stripBom(options.source);
+
+    let templateAST;
+
+    try {
+      templateAST = parse(source);
+    } catch (error) {
+      let message = buildErrorMessage(options.moduleId, error);
+      messages.push(message);
+    }
+
+    let rules = this.buildRules({
+      results: messages,
+      pending: pendingStatus,
+      moduleId: options.moduleId,
+      moduleName: options.moduleId,
+      configResolver: options.configResolver,
+      rawSource: source,
+    });
+
+    for (let rule of rules) {
+      try {
+        transform(templateAST, () => rule.getVisitor());
+      } catch (error) {
+        let message = buildErrorMessage(options.moduleId, error);
+        messages.push(message);
+      }
+    }
+
+    if (pendingStatus && messages.length === 0) {
+      messages.push({
+        message: `Pending module (\`${options.moduleId}\`) passes all rules. Please remove \`${options.moduleId}\` from pending list.`,
+        moduleId: options.moduleId,
+        severity: 2,
+      });
+    }
+
+    return messages;
+  }
+
+  logLintingError(name, moduleName, message) {
+    this._messages.push({
+      rule: name,
+      moduleId: moduleName,
+      message,
+    });
+
+    this.console.log(message);
+  }
+
+  static errorsToMessages(filePath, errors, options) {
+    errors = errors || [];
+    options = options || {
+      verbose: false,
+    };
+
+    if (errors.length === 0) {
+      return '';
+    }
+
+    let errorsMessages = errors.map(error => this._formatError(error, options)).join('\n');
+
+    return `${chalk.underline(filePath)}\n${errorsMessages}\n`;
+  }
+
+  static _formatError(error, options) {
+    let message = '';
+
+    let line = error.line === undefined ? '-' : error.line;
+    let column = error.column === undefined ? '-' : error.column;
+
+    message += chalk.dim(`  ${line}:${column}`);
+
+    if (error.severity === WARNING_SEVERITY) {
+      message += `  ${chalk.yellow('warning')}`;
+    } else {
+      message += `  ${chalk.red('error')}`;
+    }
+
+    message += `  ${error.message}  ${chalk.dim(error.rule)}`;
+
+    if (options.verbose) {
+      message += `\n${error.source}`;
+    }
+
+    return message;
+  }
+}
+
+module.exports = Linter;
+module.exports.WARNING_SEVERITY = WARNING_SEVERITY;
```

Comment:
If you make these as static class fields you don't have to export them here or in lib/index.js